src/common/mfu_daos.c

```diff
@@ -3580,24 +3676,50 @@ int daos_cont_serialize_hdlr(int rank, struct hdf5_args *hdf5, char *output_dir,
         oid.lo = mfu_flist_file_get_oid_low(flist, i);
         (*hdf5->oid)[i].oid_hi = oid.hi;
         (*hdf5->oid)[i].oid_low = oid.lo;
-        rc = daos_obj_open(da->src_coh, oid, 0, &oh, NULL);
-        if (rc != 0) {
-            MFU_LOG(MFU_LOG_ERR, "failed to open object: "DF_RC, DP_RC(rc));
-            goto out;
-        }
-        rc = serialize_dkeys(hdf5, &dk_index, &ak_index,
-                             &oh, &i, stats);
-        if (rc != 0) {
-            MFU_LOG(MFU_LOG_ERR, "failed to serialize keys: %d", rc);
-            goto out;
-        }
-        /* close source and destination object */
-        rc = daos_obj_close(oh, NULL);
-        if (rc != 0) {
-            MFU_LOG(MFU_LOG_ERR, "failed to close object: "DF_RC, DP_RC(rc));
-            goto out;
-        }
 
+        /* TODO: DAOS_OF_KV_FLAT uses daos_kv_* functions, and
+         * other object types use daos_obj_* functions. Maybe there is
+         * a better way to organize this with swtich statements, or
+         * creating "daos_obj" wrappers, etc. */
```

Comment:
@dsikich perhaps we could have open and close wrappers that check the oid features and call the corresponding kv_ or obj_ functions

Reply:
@daltonbohning yeah, I was thinking we could do that, but I am not sure if it is worth making wrappers since I don't think we will be adding more API's?

Reply:
@dsikich Yeah, maybe not. I think right now open/close is only used in like 2 different places. So there're only 2 copies of each, and it's very simple

Reply:
@dsikich But it could help simplify the if/else if we do this for dcp/dsync as well